edge-modules/edgehub-proxy/src/main.rs

```diff
@@ -126,17 +124,15 @@ fn run() -> Result<(), Error> {
         )
         .get_matches();
 
-    let mut tokio_runtime = tokio::runtime::current_thread::Runtime::new()?;
-    let url = Url::parse(
-        matches
-            .value_of("host")
-            .expect("no value for required HOST"),
-    )?;
-    let client = client(&url)?;
-
-    let api_version = matches
-        .value_of("apiversion")
-        .expect("no default value for API_VERSION");
+    let mut tokio_runtime = tokio::runtime::Builder::new_current_thread()
+        .build()?;
+
+    let url = matches
+        .value_of("host")
+        .expect("no value for required HOST");
```

Comment:
I'd prefer this code not panic. Please change this (and the expects below) to Results.